mne/decoding/time_gen.py

```diff
@@ -387,40 +388,88 @@ def _predict_time_loop(X, estimators, cv, slices, predict_mode):
+        if (len(estimators) != len(cv)) or (cv.n != n_epochs):
+            raise ValueError(
+                'When `predict_mode = "cross-validation"`, the training '
+                'and predicting cv schemes must be identical.')
+        all_test = np.concatenate(list(zip(*cv))[-1])
+        test_slices = []
+        start = 0
+        for _, test in cv:
+            l = len(test)
+            stop = start + l
+            test_slices.append(slice(start, stop, 1))
+            start += l
+
+    if all(len(s) == 1 for s in slices):
+        slices = [slice(s[0], s[0] + 1, 1) for s in slices]
+
+    y_pred = list()
+
+    # check whether the GAT is based on window length = 1-time-sample
+    # Note that we have to check i) that slices' step = 1 and ii) that slices
+    # are consecutive.
+    is_single_time_sample = True
+    expected_start = np.arange(len(slices))
+    expected_stop = expected_start + 1
+    if any(not isinstance(sl, slice) for sl in slices):
+        is_single_time_sample = False
+    elif not np.array_equal([sl.start for sl in slices], expected_start):
+        is_single_time_sample = False
+    elif not np.array_equal([sl.stop for sl in slices], expected_stop):
+        is_single_time_sample = False
+    elif slices[-1].stop != X.shape[-1]:
+        is_single_time_sample = False
+
+    msg = 'vectoring predictions across times'
+    if is_single_time_sample:
+        # in simple mode, we don't need to iterate over time slices
+        slices = [slice(expected_start[0], expected_stop[-1], 1)]
+        msg = 'not ' + msg + ', using a time window with length > 1'
+    logger.info(msg)
+
     for t, indices in enumerate(slices):
-        # Flatten features in case of multiple time samples
-        Xtrain = X[:, :, indices].reshape(
-            n_epochs, np.prod(X[:, :, indices].shape[1:]))
-
-        # Single trial predictions
-        if predict_mode == 'cross-validation':
-            # If predict within cross validation, only predict with
-            # corresponding classifier, else predict with each fold's
-            # classifier and average prediction.
-
-            # Check that training cv and predicting cv match
-            if (len(estimators) != len(cv)) or (cv.n != Xtrain.shape[0]):
-                raise ValueError(
-                    'When `predict_mode = "cross-validation"`, the training '
-                    'and predicting cv schemes must be identical.')
+        # Flatten features in case of multiple time samples given to the
+        # estimators
+        if not is_single_time_sample:
+            X_pred = X[:, :, indices].reshape(n_epochs, -1)
+        else:
+            X_pred = X
+        # import pdb; pdb.set_trace()
+
+        if predict_mode == 'mean-prediction':
+            y_pred.append(_predict(X_pred, estimators,
+                          is_single_time_sample=is_single_time_sample))
+        elif predict_mode == 'cross-validation':
+            X_t = X_pred[all_test]
```

Comment:
I'm not sure to understand this `all_test`, perhaps change the name?

Also, `X_t` could be something like `X_pred_t` to make the transition from `X`, to `X_pred`, and `X_pred_t` a bit more explicit. ? WDYT